src/mixins/Themeable.ts

```diff
@@ -254,15 +257,60 @@ export function ThemeableMixin<T extends Constructor<WidgetBase<ThemeablePropert
 		 */
 		private recalculateThemeClasses() {
 			const { properties: { theme = {} } } = this;
-			this._baseClasses = (this.getDecorator('baseClasses') || [])[0];
-			const themeKey = this._baseClasses[THEME_KEY];
-			this._baseClassesReverseLookup = createBaseClassesLookup(this._baseClasses);
-			this._theme = theme[themeKey] || {};
+			if (!this._registeredBaseThemes) {
+				this._registeredBaseThemes = this.getRegisteredBaseThemes();
+			}
+			const registeredBaseThemeKeys = this._registeredBaseThemes.map((registeredBaseThemeClasses) => {
+				return registeredBaseThemeClasses[THEME_KEY];
+			});
+
+			this._baseThemeClassesReverseLookup = createThemeClassesLookup(this._registeredBaseThemes);
+			this._theme = registeredBaseThemeKeys.reduce((baseTheme, themeKey) => {
+				return assign(baseTheme, theme[themeKey]);
+			}, {});
+
 			this._registeredClassesMap.forEach((value, key) => {
 				this.registerThemeClass(key);
 			});
 			this._recalculateClasses = false;
 		}
+
+		/**
+		 * Find the base theme class for the class name
+		 */
+		private getBaseThemeClass(className: string): string {
+			const registeredBaseTheme = find(this._registeredBaseThemes, (registeredBaseThemeClasses) => {
+				return Boolean(registeredBaseThemeClasses[className]);
+			});
+			return registeredBaseTheme[className] || '';
+		}
+
+		/**
+		 * Return the registered base themes. Warns for duplicate base theme classes.
+		 */
+		private getRegisteredBaseThemes(): ThemeClasses[] {
+			const registeredBaseThemes = this.getDecorator('baseThemeClasses').reverse();
+			registeredBaseThemes.forEach((registeredBaseThemeClasses, index) => {
+				Object.keys(registeredBaseThemeClasses).some((key) => {
+					if (key !== THEME_KEY) {
+						let duplicate = false;
+						for (let i = 0; i < registeredBaseThemes.length; i++) {
```

Comment:
if this code is purely for catching dupes, it might be cleaner to move this to a separate function which would make the whole block slightly easier to read.